scripts/kernel-crawler.py

```diff
@@ -265,6 +280,33 @@ def process_al_distro(al_distro_name, current_repo):
     else:
         return False
 
+#
+# Fedora Atomic needs 2 levels of discovery(for version, and build id, respectively)
+#
+def process_atomic_distro(current_repos):
+    for repo in current_repos["Fedora-Atomic"]:
+        try:
+            root = urllib2.urlopen(repo["root"],timeout=URL_TIMEOUT).read()
+        except:
+            continue
+        versions = html.fromstring(root).xpath(repo["version_discovery_pattern"], namespaces = {"regex": "http://exslt.org/regular-expressions"})
+        for version in versions:
+            version_url=repo["root"] + version
+            try:
+                version_page=urllib2.urlopen(version_url,timeout=URL_TIMEOUT).read()
+            except:
+                continue
+            builds = html.fromstring(version_page).xpath(repo["build_discovery_pattern"], namespaces = {"regex": "http://exslt.org/regular-expressions"})
```

Comment:
We might be able to refactor this into the generic crawling code. If it's for a distro without `build_discovery_pattern`, `builds` could take what's in `versions`.

However, if we change how we're selecting atomic kernels (in the above comment), we may still need a fedora-atomic specific function for this.

Reply:
good point. Fedora atomic packages are really odd indeed - e.g. 4.16.x alone has almost 50 releases (for x86_64). For now, fedora atomic is the only kind that requires multi discovery, plus we may want to change the discovery policy (e.g. based on publish date or white-list based ) after we gather sufficient feedback. Based on such consideration, it seems reasonable to open a task to keep track fedora atomic pre-compilation and re-evaluation of its discovery policies. How do you think?

Reply:
Yeah, we've also talked about building probes on demand, which would work better for this case.

Let's stick with this for now, and we can improve it with future probe builder changes.